[DRAFT] Documentaçao Cloudinary

marqueswill opened the pull request:
#### O que foi feito:
- [x] Pasta "Bibliotecas Auxiliares"
- [x] Link homepage para "Bibliotecas Auxiliares"
- [x] Instalaçao, criaçao da nuvem e configuraçao
- [x] Botao de upload
- [x] CldImg

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
```diff
@@ -0,0 +1,219 @@
+
+TypeScript é uma extensão da linguagem JavaScript. Em suma, ele traz definições de tipos para o seu código, a fim de se evitar alguns tipos de bugs e diversas depurações relacionadas tanto com a passagem de parâmetros quanto com a validação de que objetos, estados e variáveis irão se comportar como você espera.
+
+Vale reiterar que o TypeScript não é uma nova linguagem. No final, todo seu código irá ser convertido em JavaScript, para quando for lido pelo compilador.
```
```suggestion
Vale reiterar que o TypeScript não é uma nova linguagem. No final, todo seu código irá ser transpilado (convertido) em JavaScript, para poder ser interpretado.
```

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
```diff
@@ -0,0 +1,219 @@
+---
+
+# De .jsx para .tsx
+
```
Acho bomm ter um: Pra quê usar typescript?

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
Acho melhor separar o typescript pra React em outro arquivo, e ter uma abordagem menos ligada a framework inicialmente.

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
````diff
@@ -0,0 +1,219 @@
+O hook useState é capaz de realizar inferência de tipos, a partir do valor inicial passado na declaração do estado.
+
+```
+const [state, setState] = useState("string")
````
```suggestion
const [state, setState] = useState("valor inicial")
```

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
```diff
@@ -0,0 +1,219 @@
+type Theme = "light" | "dark" | "system";
+const ThemeContext = createContext<Theme>("system");
+
+const useGetTheme = () => useContext(ThemeContext);
```
```suggestion
const useThemeContext = () => useContext(ThemeContext);
```

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
```diff
@@ -0,0 +1,219 @@
+      <input value={email} placeholder="email" onChange={handleEmail}/>
+      <input value={password} placeholder="password" onChange={handlePassword}/>
+      <button type="submit"/>
+    <form/>
```
```suggestion
    </form>
```

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
````diff
@@ -0,0 +1,219 @@
+* Para onSubmit de um form, use FormEvent
+
+```
+import { useState, type FormEvent } from 'react';
````
```suggestion
import {
  useState,
  type FormEvent,
  type ChangeEvent,
  type FormEventHandler,
} from "react";
```

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
```diff
@@ -0,0 +1,219 @@
+    event.preventDefault();
+    console.log(email, password);
+  }
+  
```
Acho que seria massa mostrar outra alternativa
```suggestion
  const handleSubmitAlternative: FormEventHandler<HTMLFormElement> = (
    event // tipo do argumento já está especificado pelo tipo da função
  ) => {
    event.preventDefault();
  };
```

artistrea commented on Diretoria Projetos/stack_T3/typescript/explicacao.md:
````diff
@@ -0,0 +1,219 @@
+
+No @types/react, existem [diversos tipos](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L163C1-L163C1) para componentes e marcação do nosso código jsx, o que pode acabar confundindo quem está vendo TypeScript pela primeira vez. Portanto, vamos nos focar nos tipos básicos, que podemos usar para a maioria das situações, como para passagem de componentes em pr ...
+
+Tipando componentes:
+
+* ReactNode
+
+```
+interface MyComponentProps {
+  title: string;
+  children: ReactNode;
+}
+```
+
+Uma primeira possibilidade é usar o tipo ReactNode, o qual é uma união de todos os possíveis tipos que podem ser passados como children em um código JSX.
+
+* ReactElement
+
+```
+interface MyComponentProps {
+  title: string;
+  children: ReactElement;
+}
+```
+
+Uma segunda possibilidade, menos abrangente, é utilizar o ReactElement, o qual abrange apenas elementos JSX, excluindo tipos primitivos do JavaScript como string e number.
+
+* HTMLElement
+
+```
+interface MyComponentProps {
+  title: string;
+  children: HTMLElement
+}
+```
````
Acho bem mais fácil usar isto, ao invés de ficar lembrando qual é qual. Eu colocaria como primeiro exemplo esse daqui:
```
type MyComponentProps = React.PropsWithChildren<{
    title: string
}>
```

artistrea reviewed (changes requested):
Como isso não é diretamente relacionado com o T3 stack, acho bom deixar separado. Tiraria também da Diretoria Projetos.
 - [x] Colocar documentação numa pasta como ´doc. técnica/typescript´ ou `execução/typescript`;
 - [x] Separar typescript puro do React TS;

Além disso, é super arriscado ter numa documentação para quem tá iniciando em typescript esse tanto de exemplo com interface. Interface tem um caso de uso muito específico (biblioteca e possibilitar estender um tipo depois de sua declaração inicial). Sugiro citar os dois, suas difereças, o perigo de interface, se quiser falar da tua preferência, pode, e usar só type nos exemplos.

- [x] A não ser que tenha uma razão além de preferência, coloca type nos exemplos

artistrea commented:
Sobre as Possíveis alterações:

- adicionar tipagem para useReducer
- adicionar como manipular e tipar funções async (dá bug se passar em um onClick)
- explicar DOMElement

useReducer acho que não (nunca usamos),
Acho que pode valer a pena falar sobre Promise,
e DOMElement nem faço ideia de pra que serve

artistrea commented on Execucao/typescript/explicacao.md:
```diff
@@ -0,0 +1,267 @@
+
+Se você já programou em JavaScript, torna-se ainda mais fácil programar em TypeScript, afinal eles são basicamente "a mesma coisa". O que isso quer dizer? É que o TypeScript é na verdade um superset open-source, desenvolvido pela Microsoft, ou seja, uma extensão da linguagem JavaScript, com a adição de algumas propriedades que melhoram o nosso código.
+
+Tá, mas ainda não chegamos no tópico principal desta introduçaço né... por que usar o TypeScript? Para começar, ele funciona como uma ferramenta que adiciona tipagem estática ao JavaScript, que pode ser escrita em qualquer ambiente de desenvolvimento e que, quando instalada via gerenciador de pacotes JS, permite checar erros e utilizar outros compiladores que suportam este mecanismo. Com isso, o  ...
```
introduçaço

artistrea commented on Execucao/typescript/explicacao.md:
```diff
@@ -0,0 +1,267 @@
+
+Tá, mas ainda não chegamos no tópico principal desta introduçaço né... por que usar o TypeScript? Para começar, ele funciona como uma ferramenta que adiciona tipagem estática ao JavaScript, que pode ser escrita em qualquer ambiente de desenvolvimento e que, quando instalada via gerenciador de pacotes JS, permite checar erros e utilizar outros compiladores que suportam este mecanismo. Com isso, o  ...
+Além do potencial de detecção de erros durante o desenvolvimento de projetos, outra vantagem é a possibilidade de incluir a IntelliSense da IDE que está sendo usada. Isso reflete num ambiente muito mais ágil e seguro enquanto o código está sendo digitado pelo desenvolvedor.
+
```
Eu explicaria o que quer dizer por seguro, citando a segurança e facilidade de manutenção e modificação do código

pafev replied:
Não entendi, a explicação do "por que seguro" já não está no subtópico "Verificador de Tipos Estáticos"?

artistrea commented on Execucao/typescript/explicacao.md:
````diff
@@ -0,0 +1,267 @@
+
+```
+const verificaPositivo: (num: number) => boolean = (num) => {
+    if (num >= 0) {
+        return true;
+    }
+    return false;
````
Quem nunca fez isso? kkkkkk
```suggestion
    return num >= 0;
```

artistrea commented on Execucao/typescript_react/explicacao.md:
```diff
@@ -0,0 +1,238 @@
+
+Por fim, se você precisar usar algum evento que não está incluído nessa lista, você pode usar o tipo SyntheticEvent, que é o tipo do qual é extendido todos os outros tipos de evento.
+
+### Componentes (ReactNode, ReactElement, HTMLElement)
```
Acho que esse trecho devia ser subtópico de Tipando Props, não?

pafev replied:
hmmm aí eu teria que fazer uma divisão diferente, porque este subtópico e o de props de estilização utilizam coisas do @types/react, e por isso tão neste tópico em que estão. Aí eu teria que fazer uma divisão do tipo

pafev replied:
Ou seja, seria mover o Tipando Props pra dentro do tópico Pacote @types/react
Você acha que assim ficaria melhor?

artistrea commented on Execucao/typescript_react/explicacao.md:
```diff
@@ -0,0 +1,238 @@
+
+Mais uma possibilidade, ainda menos abrangente, é utilizar o HTMLElement, o qual abrange apenas marcações do HTML, excluindo componentes React e tipos primitivos do JavaScript.
+
+### Props de estilização
```
Acho que esse trecho também devia ser subtópico de Tipando Props, não?

artistrea reviewed (changes requested):
- [x] Colocar as coisas de typescript em react dentro da pasta de typescript;
- [ ] Aumentar a quantidade de metadados e melhorar os ícones usados;
    - Use como base a [PR](https://github.com/StructCE/DocStruct/pull/15/files#diff-7b2fd61a372cb243593e60d78f8bbb220ad9652df3c4cae161bd812f6d71a9ef), que segue as dicas dadas pelo Araújo
- [x] Só deve existir um H1 por página, em `explicação` tem alguns H1 diferentes.
- [ ] Uma dica pra ficar melhor de acompanhar textos maiores: chamar atenção para as palavras e frases chave usando negrito, alertas, etc:
    - Usa \*\*em volta da palavra\*\* pra ficar assim: **em volta da palavra**
- [x] Colocar o nome da linguagem no bloco de código, para ter *syntax highlight*:
    - É assim que se inicia um bloco de código tsx, por exemplo \`\`\`tsx
- [ ] Tá faltando como adicionar typescript a um projeto, e um pouco sobre o `tsconfig`


Essa documentação ficou muito bem escrita!

marqueswill committed:
Primeiros passos

marqueswill committed:
documentação widget upload
pasta "bibliotecas auxiliares"
configurações página inicial

marqueswill committed:
Fotitas de autores
Documentação CldImage
Inclusão do publicId no upload
Reordenação dos docs da stack

marqueswill commented:
Dei uma organizadinha na ordem das tecnologias pra ficar algo mais progressivo
![Captura de tela de 2024-04-05 15-38-39](https://github.com/StructCE/DocStruct/assets/100526142/c7256446-2ddf-4722-9303-2705b9a22f4f)
